<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>main_video_player_window</class>
 <widget class="QMainWindow" name="main_video_player_window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1388</width>
    <height>664</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1024</width>
    <height>480</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Video Loop</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="widget_central">
   <layout class="QGridLayout" name="layout_main">
    <item row="1" column="1">
     <widget class="QLineEdit" name="entry_video">
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="2" column="0" colspan="4">
     <widget class="TimestampTableView" name="list_timestamp">
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SingleSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
      <attribute name="horizontalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QPushButton" name="button_video_browse">
      <property name="text">
       <string>Browse</string>
      </property>
     </widget>
    </item>
    <item row="2" column="5" rowspan="3">
     <widget class="QFrame" name="frame_media">
      <property name="minimumSize">
       <size>
        <width>500</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="1" column="0">
        <layout class="QVBoxLayout" name="media_layout">
         <property name="sizeConstraint">
          <enum>QLayout::SetNoConstraint</enum>
         </property>
         <item>
          <widget class="VideoFrame" name="frame_video">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="focusPolicy">
            <enum>Qt::StrongFocus</enum>
           </property>
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Plain</enum>
           </property>
           <property name="lineWidth">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="frame_media_control">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>70</height>
            </size>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Plain</enum>
           </property>
           <property name="lineWidth">
            <number>0</number>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="1" column="15">
             <widget class="QToolButton" name="btnRight">
              <property name="toolTip">
               <string>Step Frames Right</string>
              </property>
              <property name="text">
               <string>&gt;</string>
              </property>
              <property name="shortcut">
               <string>Right</string>
              </property>
              <property name="arrowType">
               <enum>Qt::RightArrow</enum>
              </property>
             </widget>
            </item>
            <item row="1" column="9">
             <spacer name="horizontalSpacer_7">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="1" column="10">
             <widget class="QPushButton" name="button_full_screen">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Set the video to full screen</string>
              </property>
              <property name="text">
               <string>Full Screen</string>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QDoubleSpinBox" name="doubleSpinBoxPlaybackSpeed">
              <property name="minimum">
               <double>0.200000000000000</double>
              </property>
              <property name="maximum">
               <double>6.000000000000000</double>
              </property>
              <property name="value">
               <double>1.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="8">
             <widget class="QPushButton" name="button_mark_end">
              <property name="toolTip">
               <string>Mark the end of the entry</string>
              </property>
              <property name="text">
               <string>Mark End</string>
              </property>
             </widget>
            </item>
            <item row="1" column="6">
             <spacer name="horizontalSpacer_6">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="1" column="0">
             <widget class="ToggleButton" name="button_play_pause">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Toggle Play/Pause</string>
              </property>
              <property name="text">
               <string>Play/Pause</string>
              </property>
              <property name="checkable">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item row="1" column="7">
             <widget class="QPushButton" name="button_mark_start">
              <property name="toolTip">
               <string>Mark the start of the entry</string>
              </property>
              <property name="text">
               <string>Mark Start</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="19">
             <widget class="HighlightedJumpSlider" name="slider_progress">
              <property name="maximum">
               <number>9999</number>
              </property>
              <property name="pageStep">
               <number>1</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::NoTicks</enum>
              </property>
             </widget>
            </item>
            <item row="1" column="11">
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="1" column="14">
             <widget class="QSpinBox" name="spinBoxFrameJumpMultiplier">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
             </widget>
            </item>
            <item row="1" column="17">
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="1" column="13">
             <widget class="QToolButton" name="btnLeft">
              <property name="toolTip">
               <string>Step Frames Left</string>
              </property>
              <property name="text">
               <string>&lt;</string>
              </property>
              <property name="shortcut">
               <string>Left</string>
              </property>
              <property name="arrowType">
               <enum>Qt::LeftArrow</enum>
              </property>
             </widget>
            </item>
            <item row="1" column="12">
             <widget class="QToolButton" name="btnSkipLeft">
              <property name="toolTip">
               <string>Skip Frames Left</string>
              </property>
              <property name="text">
               <string>&lt;-</string>
              </property>
              <property name="shortcut">
               <string>Ctrl+Left</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <spacer name="horizontalSpacer_5">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="1" column="16">
             <widget class="QToolButton" name="btnSkipRight">
              <property name="toolTip">
               <string>Skip Frames Right</string>
              </property>
              <property name="text">
               <string>-&gt;</string>
              </property>
              <property name="shortcut">
               <string>Ctrl+Right</string>
              </property>
             </widget>
            </item>
            <item row="1" column="18">
             <widget class="QPushButton" name="btnHelp">
              <property name="text">
               <string>Help</string>
              </property>
             </widget>
            </item>
            <item row="1" column="4">
             <widget class="QPushButton" name="button_speed_up">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Speed up the video</string>
              </property>
              <property name="text">
               <string>Speed Up</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QPushButton" name="button_slow_down">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Slow down the video</string>
              </property>
              <property name="text">
               <string>Slow Down</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="0" colspan="4">
     <widget class="QFrame" name="frame_timestamp_detail">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::Panel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QFormLayout" name="formLayout">
       <property name="labelAlignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label_start_time">
         <property name="text">
          <string>Start Time</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="entry_start_time">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_end_time">
         <property name="text">
          <string>End Time</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="entry_end_time">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_description">
         <property name="text">
          <string>Description</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="PlainTextEdit" name="entry_description">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>75</height>
          </size>
         </property>
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="2">
        <widget class="QFrame" name="frame">
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Plain</enum>
         </property>
         <property name="lineWidth">
          <number>0</number>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="button_add_entry">
            <property name="text">
             <string>Add Entry</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="button_remove_entry">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Remove Entry</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="button_save">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="button_run">
            <property name="text">
             <string>Run</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="3">
     <widget class="QPushButton" name="button_timestamp_browse">
      <property name="text">
       <string>Browse</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLineEdit" name="entry_timestamp">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="0" column="4">
     <widget class="QPushButton" name="button_timestamp_create">
      <property name="text">
       <string>Create</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_video">
      <property name="text">
       <string>Video File</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="label_timestamp">
      <property name="text">
       <string>Timestamp File</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="5">
     <widget class="QWidget" name="widget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="topMargin">
        <number>9</number>
       </property>
       <property name="bottomMargin">
        <number>4</number>
       </property>
       <item>
        <widget class="QLabel" name="lblVideoName">
         <property name="font">
          <font>
           <pointsize>12</pointsize>
          </font>
         </property>
         <property name="text">
          <string>VideoName</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="lblVideoSubtitle">
         <property name="text">
          <string>Subtitle</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QDateTimeEdit" name="dateTimeEdit">
         <property name="enabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="5">
     <widget class="QWidget" name="widget_2" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <property name="spacing">
        <number>6</number>
       </property>
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="title">
          <string>Other</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <property name="topMargin">
           <number>2</number>
          </property>
          <property name="bottomMargin">
           <number>2</number>
          </property>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>FPS:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lblFileFPS">
            <property name="text">
             <string>--</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Playback Percent:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lblPlaybackPercent">
            <property name="text">
             <string>--</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Playback Time</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <property name="topMargin">
           <number>2</number>
          </property>
          <property name="bottomMargin">
           <number>2</number>
          </property>
          <item>
           <widget class="QLabel" name="lblCurrentTime">
            <property name="text">
             <string>--</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lblTotalDuration">
            <property name="text">
             <string>--</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Video Frames</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <property name="topMargin">
           <number>2</number>
          </property>
          <property name="bottomMargin">
           <number>2</number>
          </property>
          <item>
           <widget class="QSpinBox" name="spinBoxCurrentFrame">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::UpDownArrows</enum>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>10000</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lblCurrentFrame">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>--</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lblTotalFrames">
            <property name="text">
             <string>--</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionTest_Entry">
   <property name="text">
    <string>Test Entry</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>VideoFrame</class>
   <extends>QFrame</extends>
   <header>GUI.UI.VideoPlayer</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>HighlightedJumpSlider</class>
   <extends>QSlider</extends>
   <header>GUI.UI.VideoPlayer</header>
  </customwidget>
  <customwidget>
   <class>TimestampTableView</class>
   <extends>QTableView</extends>
   <header>GUI.UI.VideoPlayer</header>
  </customwidget>
  <customwidget>
   <class>PlainTextEdit</class>
   <extends>QPlainTextEdit</extends>
   <header>GUI.UI.VideoPlayer</header>
  </customwidget>
  <customwidget>
   <class>ToggleButton</class>
   <extends>QPushButton</extends>
   <header>GUI.UI.VideoPlayer</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
